content_sync/apis/github.py

```diff
@@ -137,18 +139,23 @@ def get_app_installation_id(app: GithubIntegration) -> Optional[str]:
 def get_token():
     """ Get a github token for requests """
     if settings.GITHUB_APP_ID and settings.GITHUB_APP_PRIVATE_KEY:
-        app = GithubIntegration(
-            settings.GITHUB_APP_ID,
-            default_backend().load_pem_private_key(
-                settings.GITHUB_APP_PRIVATE_KEY, None
-            ),
-            **(
-                {"base_url": settings.GIT_API_URL}
-                if settings.GIT_API_URL is not None
-                else {}
-            ),
-        )
-        return app.get_access_token(get_app_installation_id(app)).token
+        try:
+            app = GithubIntegration(
+                settings.GITHUB_APP_ID,
+                default_backend().load_pem_private_key(
+                    settings.GITHUB_APP_PRIVATE_KEY, None
+                ),
+                **(
+                    {"base_url": settings.GIT_API_URL}
+                    if settings.GIT_API_URL is not None
+                    else {}
+                ),
+            )
+            return app.get_access_token(get_app_installation_id(app)).token
+        except Exception as exc:
+            raise ImproperlyConfigured(
```

Comment:
I wonder if there's an exception other than bad credentials, the error message will still be to check credentials?
Though I'm not sure if this is something to be taken care of as we are already raising `ImproperlyConfigured`

Reply:
If anything goes wrong here it's most likely going to be due to wrong settings used to authenticate with git (`GITHUB_APP_ID`, `GITHUB_APP_PRIVATE_KEY`, `GIT_API_URL`).  How about if I change the message to `Could not initialize github app, check the relevant settings`?